lib/facter/env_temp_variable.rb

```diff
@@ -0,0 +1,3 @@
+Facter.add(:env_temp_variable) do
+  setcode { ENV['TEMP'] }
```

Comment:
FYI, talking about cases where it doesn't resolve to a value. And they are far and few but should be handled.